### PR TITLE
DLT upstream fixes and new features

### PR DESCRIPTION
**Pull request by minminlittleshrimp**
Upstreamming activity

### CODE REVIEW

File: tests/CMakeLists.txt
Comment (lvklevankhanh) [inline tests/CMakeLists.txt:1]: It seem missing the configure_file(). 
Please replace:
-configure_file(${PROJECT_SOURCE_DIR}/tests/testfile.dlt ${PROJECT_BINARY_DIR}/tests COPYONLY)
-configure_file(${PROJECT_SOURCE_DIR}/tests/testfilter.txt ${PROJECT_BINARY_DIR}/tests COPYONLY)
-configure_file(${PROJECT_SOURCE_DIR}/tests/testfile_filetransfer.txt ${PROJECT_BINARY_DIR}/tests COPYONLY)
-configure_file(${PROJECT_SOURCE_DIR}/tests/test_dlt.conf ${PROJECT_BINARY_DIR}/tests COPYONLY)

to:
+configure_file(${PROJECT_SOURCE_DIR}/tests/testfile.dlt ${CMAKE_CURRENT_BINARY_DIR} COPYONLY)
+configure_file(${PROJECT_SOURCE_DIR}/tests/testfilter.txt ${CMAKE_CURRENT_BINARY_DIR} COPYONLY)
+configure_file(${PROJECT_SOURCE_DIR}/tests/testfile_filetransfer.txt ${CMAKE_CURRENT_BINARY_DIR} COPYONLY)
+configure_file(${PROJECT_SOURCE_DIR}/tests/test_dlt.conf ${CMAKE_CURRENT_BINARY_DIR}/tests COPYONLY)

Comment (minminlittleshrimp) [reply tests/CMakeLists.txt:1]: Hello @lvklevankhanh 
Please kindly check, I already fix the patch.
However, due to some not-yet-upstream commits, there is no test_dlt.conf, hence I excluded the last line:
```
configure_file(${PROJECT_SOURCE_DIR}/tests/test_dlt.conf ${CMAKE_CURRENT_BINARY_DIR}/tests COPYONLY)
```
Regards

**Comment by minminlittleshrimp** [comment]
Set to DRAFT until reworking done.
Regards

**Comment by minminlittleshrimp** [comment]
Closed due to some unclear commit issues

**Pull request closed without merging**